src/client.c

```diff
@@ -1754,8 +1769,10 @@ int main(int argc, char *argv[]) {
         { "host", 't', 0, G_OPTION_ARG_STRING_ARRAY, &hostarr,
             "Set host for a recipe with specific id.",
             "<recipe_id>=[<user>@]<host>" },
-        {"conn-retries", 'c', 0, G_OPTION_ARG_INT, &app_data->max_retries,
+        { "conn-retries", 'c', 0, G_OPTION_ARG_INT, &app_data->max_retries,
             "Specify the number of reconnection retries.", NULL},
+        { "port", 'p', 0, G_OPTION_ARG_INT, &app_data->restraint_port,
```

Comment:
Not sure where my inline comment went but here goes again.
If you are striving to be backward compatible, previously the port used to be provided by way as host argument
and 8082 was the port number:
--host 1=root@dev-kvm-guest-07.rhts.eng.bos.redhat.com:8082

Reply:
I opened #28 and I asked for review there to address this type of concerns before code changes.

> 0. Recover "-p, --port" option in server and client. If the option is not used, keep the dynamic port behavior.

https://github.com/beaker-project/restraint/blob/2b41f95b00904ea73ed40fa605c6810bd892bd8b/src/client.c#L1666

Quoting myself on a resolved comment on this PR:

> The purpose of the option is to allow the service to use the same port across reboots, and it's added to the client in case that some workflow running standalone needs a specific port. Otherwise, for standalone, port is printed out while running. So not sure if it's really needed to added everywhere.

The host argument changed two times since 0.1.45, and it is reflecting the current standalone behavior, where the client is starting restraintd on the recipe target host through ssh. As it is right now, the host argument is for the ssh connection.

- 0.1.45: `<recipe_id>=[<user>@]<host>[:<port>][/ssh_port]`
- 0.2.0:   `<recipe_id>=[<user>@]<host>[:ssh_port]`
- master: `<recipe_id>=[<user>@]<host>`

The format in 0.2.0 is not valid, as you cannot use host:port syntax with the ssh client.

Taking port for restraintd from host argument, as in 0.1.45, will not help to not break things, as in 0.1.45, the client expected restraintd to be already running in that port. Current client will fail because it will try to start a restraintd process listening on a port that is already being used by other restraintd instance.
This is related to https://bugzilla.redhat.com/show_bug.cgi?id=1823545 and users are OK with moving their tasks to 0.2.0 standalone behavior.

And note that the default port for the client in 0.1.45 was 8081,

https://github.com/beaker-project/restraint/blob/2b41f95b00904ea73ed40fa605c6810bd892bd8b/src/ssh.h#L8

https://github.com/beaker-project/restraint/blob/2b41f95b00904ea73ed40fa605c6810bd892bd8b/src/ssh.c#L262

In any case, this is discussion for #12.

Reply:
Only reason I suggested this as you seem to be concerned about being backward compatible.  Code could have easily extracted substrings from --host.  But I guess in this case it isn't important to be backward compatible.